src/components/ToastNotification/ToastNotification.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { FC, useEffect, useState, useRef, useCallback } from 'react';
+
+import { Button } from '../Button';
+import { IconInfoCircle, IconXClose } from '../Icons';
+
+import * as Toast from '@radix-ui/react-toast';
+
+import styles from './ToastNotification.module.scss';
+
+type ToastPositionVariant = 'left' | 'right';
+type ToastThemeVariant = 'primary' | 'success' | 'error';
+
+const { ToastRoot, ToastIcon, TextWrapper, ToastTitle, ToastDescription, ToastButton, ToastClose, ToastViewport } =
+  styles;
+
+export type ToastNotificationProps = Toast.ToastProviderProps & {
+  title: string;
+  description: string;
+  actionTitle: string;
+  actionAltText: string;
+  positionVariant?: ToastPositionVariant;
+  themeVariant?: ToastThemeVariant;
+  swipeDirection?: Toast.SwipeDirection;
+  openToast?: boolean;
+  onClick?: () => void;
+  onClose?: () => void;
+};
+
+export const ToastNotification: FC<ToastNotificationProps> = ({
```

Comment:
Remove `FC` here as it will add `children` as an implicit prop. 

```tsx
export const ToastNotification = ({ ... }: ToastNotificationProps) => ({ ... });
```